src/subclass/mod.rs

```diff
@@ -0,0 +1,161 @@
+//! impl SimpleObject {
+//!     // This macro defines a
+//!     //   fn get_type() -> glib::Type
+//!     // that registers the new type with the type system
+//!     // on the first call and returns its type id
+//!     glib_object_get_type!();
+//! }
+//!
+//! // ObjectSubclass is the trait that defines the new type and
+//! // contains all information needed by the GObject type system,
+//! // including the new type's name, parent type, etc.
+//! impl ObjectSubclass for SimpleObject {
+//!     // This type name must be unique per process
+//!     const NAME: &'static str = "SimpleObject";
+//!
+//!     // The parent type this one is inheriting from
+//!     type ParentType = glib::Object;
+//!
+//!     // The C/FFI instance and class structs. The simple ones
+//!     // are enough in most cases and more is only needed to
+//!     // expose public instance fields to C APIs or to provide
+//!     // new virtual methods for subclasses of this type
+//!     type Instance = subclass::simple::InstanceStruct<Self>;
+//!     type Class = subclass::simple::ClassStruct<Self>;
+//!
+//!     // This mcaro defines some boilerplate
+//!     glib_object_subclass!();
+//!
+//!     // Called right before the first time an instance of the new
+//!     // type is created. Here class specific settings can be performed,
+//!     // including installation of properties and registration of signals
+//!     // for the new type
+//!     fn class_init(klass: &mut subclass::simple::ClassStruct<Self>) {
+//!         klass.install_properties(&PROPERTIES);
+//!     }
+//!
+//!     // Called every time a new instance is created. This should return
+//!     // a new instance of our type with its basic values
+//!     fn new() -> Self {
+//!         Self {
+//!             name: RefCell::new(None),
+//!         }
+//!     }
+//! }
+//!
+//! // Trait that is used to override virtual methods of glib::Object
+//! impl ObjectImpl for SimpleObject {
+//!     // This macro defines some boilerplate
+//!     glib_object_impl!();
+//!
+//!     // Called whenever a property is set on this instance. The id
+//!     // is the same as the index of the property in the PROPERTIES array
+//!     fn set_property(&self, _obj: &glib::Object, id: usize, value: &glib::Value) {
+//!         let prop = &PROPERTIES[id];
+//!
+//!         match *prop {
+//!             subclass::Property("name", ..) => {
+//!                 let name = value.get();
+//!                 self.name.replace(name.clone());
+//!             }
+//!             _ => unimplemented!(),
+//!         }
+//!     }
+//!
+//!     // Called whenever a property is retrieved from this instance. The id
+//!     // is the same as the index of the property in the PROPERTIES array
+//!     fn get_property(&self, _obj: &glib::Object, id: usize) -> Result<glib::Value, ()> {
+//!         let prop = &PROPERTIES[id];
+//!
+//!         match *prop {
+//!             subclass::Property("name", ..) => Ok(self.name.borrow().clone().to_value()),
```

Comment:
Interesting, it works without `clone` ? Examples seems fine without it.

Reply:
Indeed unneeded, thanks!